src/modules/dapp/services/fuel-transaction.ts

```diff
@@ -40,20 +40,21 @@ export enum IFuelTransactionNames {
   TRANSFER_ASSET = 'Transfer asset',
 }
 
-// export declare enum AddressType {
-//   contract = 0,
-//   account = 1,
-// }
-
 class FuelTransactionService {
-  static async simulate({
-    providerUrl,
-    transactionLike,
-    from,
-  }: TransactionSimulateParams) {
-    const provider = await Provider.create(providerUrl);
-    const transactionRequest = ScriptTransactionRequest.from(transactionLike);
-    const fee = await provider.getTransactionCost(transactionRequest);
```

Comment:
The `provider.getTransactionCost(transactionRequest);` most of the time was returning the error `InsufficientMaxFee`.
Using the `BaseTransfer.prepareTransaction` was the solution.